velox/functions/prestosql/registration/DataSizeFunctionsRegistration.cpp

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/Registerer.h"
+#include "velox/functions/prestosql/DataSizeFunctions.h"
+
+namespace facebook::velox::functions {
+namespace {
+
+void registerSimpleFunctions(const std::string& prefix) {
+  registerFunction<ParsePrestoDataSizeFunction, int128_t, Varchar>(
```

Comment:
check this PR out for all you need to know about decimal types and simple functions (the description is pretty good) https://github.com/facebookincubator/velox/pull/9096

You need to specify LongDecimal as output type and also specify a constraint since this function only outputs a decimal type with a specific precision and scale

Finally in the simple function definition, you need to use the correct out_type like  `out_type<LongDecimal<P1, S1>>`